### ISSUE TITLE
`date` is not a valid type

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#9916]
### Describe the bug

`datetime` has been replaced by `date` it appears.

```nushell
> ls | describe
table<name: string, type: string, size: filesize, modified: date> (stream)
```
```nushell
> ls | get modified.0 | describe
date
```
```nushell
> date now | describe
date
```

however, `date` does not appear to be a valid type in the parser

### How to reproduce

run either one of the following blocks
```nushell
> let foo: date = date now
Error: nu::parser::unknown_type

  × Unknown type.
   ╭─[entry #7:1:1]
 1 │ let foo: date = date now
   ·          ──┬─
   ·            ╰── unknown type
   ╰────
```
```nushell
> let foo: date = ls | get modified.0
Error: nu::parser::unknown_type

  × Unknown type.
   ╭─[entry #8:1:1]
 1 │ let foo: date = ls | get modified.0
   ·          ──┬─
   ·            ╰── unknown type
   ╰────
```
```nushell
> def foo []: nothing -> date { ls | get modified.0 }
Error: nu::parser::unknown_type

  × Unknown type.
   ╭─[entry #12:1:1]
 1 │ def foo []: nothing -> date { ls | get modified.0 }
   ·                        ──┬─
   ·                          ╰── unknown type
   ╰────
```
also, `datetime` is still there for some reason :thinking: 

### Expected behavior

i expected the commands above to parse, as the following does
```nushell
let foo: datetime = date now
```

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.83.2                                   |
| branch             | main                                     |
| commit_hash        | a98b3124c52262d0a72c4d009cfc86ee0977bb0d |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16)      |
| rust_channel       | 1.69.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12)      |
| build_time         | 2023-08-04 09:50:54 +02:00               |
| build_rust_channel | release                                  |
| allocator          | standard                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  | hist                                     |

**Comment by amtoine** [comment]
a few additional things
- `let foo: datetime = (ls | get modified.0)` works
- but `let foo: datetime = ls | get modified.0` doesn't
```nushell
Error: nu::parser::expected_keyword

  × Expected keyword.
   ╭─[entry #19:1:1]
 1 │ let foo: datetime = ls | get modified.0
   ·                   ──────────┬──────────
   ·                             ╰── expected =
   ╰────
```
- `into datetime` might need to be renamed to `into date`?
- the `datetime` type might need to be removed?

**Comment by fdncred** [comment]
I went on a rant about this a while back and may even have opened an issue, because there are many places in the code that use date and datetime interchangeably when, imo, they're clearly two different things.

https://github.com/nushell/nushell/issues/9090#issuecomment-1533245553

**Comment by amtoine** [comment]
> I went on a rant about this a while back and may even have opened an issue (didn't find one), because there are many places in the code that use date and datetime interchangeably when, imo, they're clearly two different things.
> 
> [#9090 (comment)](https://github.com/nushell/nushell/issues/9090#issuecomment-1533245553)
> 
ooh yeah it's true i remember a conversation like that :ok_hand: 

> our representation is a datetime, so i'd vote for consolidating on datetime versus date.
>
i'm not sure i get why `datetime` and not just `date` :confused: 

we would then have
- `Value::Date` instead of `Value::Datetime`
- `date` as a type instead of `datetime`
- `date ...` for the commands, as it is now
- `into date` instead of `into datetime`

if there is a reason for the "time" of "datetime", we could also go the other way :+1: 

but definitely not both :innocent:

**Comment by fdncred** [comment]
> i'm not sure i get why datetime and not just date

Because a date is some formulation of Month, Day and Year only. A datetime has all that stuff but also includes Hour, Minutes, Seconds, etc.

Some systems have Date, Time, and DateTime. I think we should just use DateTime.

**Comment by amtoine** [comment]
> > i'm not sure i get why datetime and not just date
> 
> Because a date is some formulation of Month, Day and Year only. A datetime has all that stuff but also includes Hour, Minutes, Seconds, etc.
> 
> Some systems have Date, Time, and DateTime. I think we should just use DateTime.
>
ooh but of course it's a date and a time, thus a datetime :laughing: 
it's perfect the way you said it, i'm convinced to keep "datetime" then :blush:

**Comment by amtoine** [comment]
and yeah, only DateTime, no Date, no Time :wink:

**Comment by agirardeau** [comment]
Why no separate Date and DateTime? It's very common to handle dates without specifying time or timezone, hence why other languages have separate types.

https://github.com/nushell/nushell/issues/3205 called out that you can't create a date without specifying time zone, which seems like a pretty big failure of nushell right now.

**Comment by amtoine** [comment]
yeah i agree sometimes i just want the date and not the time, and end up setting the time to 00:00:00, either manually or by default because `2000-01-01T00:00:00` is the same as `2000-01-01` :+1:

**Comment by kgadek** [comment]
> yeah i agree sometimes i just want the date and not the time, and end up setting the time to 00:00:00, either manually or by default because `2000-01-01T00:00:00` is the same as `2000-01-01` 👍

I'm doing the same, but there's always a risk that `2000-01-01T00:00:00` could be interpreted as "in UTC"; thus being read as `1999-12-31T23:00:00 UTC-1`.

**Comment by sholderbach** [comment]
The reason why we don't have a `date` only type and just `datetime` is that there is an explicit need for the `datetime` timestamps in a bunch of Nushell provided code like modified file times in `ls` etc. Here you want information that is uniquely identifying down to fractions of a second and has to take into account the existence of timezones (with some complexity due to the system). Around this type a particular need for the (misnamed?) `date` utility commands arose to permit useful operations on this type.

A date-only type may be useful to do everyday business, but will require additional properly defined semantics and ways to interact with our indispensable `datetime` type. So it is primarily a question of complexity management (some added complexity in use or in the overall specification of the Nushell langauge and its standard types) and how sound it is to make simplifying assumptions/implicitness etc.
Still there are additional real world complexities: how do you define a date independent from timezones to agree in a world where timezones exist?

**Comment by NotTheDr01ds** [comment]
In reading through this, I think where we landed is that the issue is simply that `describe` should return `datetime` rather than `date`, right?

We're all in agreement that the correct type name is `datetime`, but `describe` doesn't use the correct name. If that's correct, can we edit the issue title to reflect that?

@amtoine - The issue that you mentioned in [this comment](https://github.com/nushell/nushell/issues/9916#issuecomment-1665410816) seems to have been fixed.  Both of those assignments now work correctly.
